### ISSUE TITLE
When creating a PolymerTemplate, I want to use POJOs as model objects, so I can reuse those POJOs inside my application.

### ISSUE THREAD

**Issue by gilberto-torrezan** [vaadin/flow#1681]
Currently the model must be defined as an interface. For example:

```
@Tag("my-component")
@HTMLImport("frontend://components/my-component.html")
public class MyComponent extends PolymerTemplate<MyModel> {
    public interface MyModel extends extends TemplateModel {
        void setProperty(String property);
    }
}
```
That's not a problem when you have a small number of properties to set/get, but when you have a long list of properties, creating a model interface becomes cumbersome. In some cases you end up having a POJO "interface" - a interface with several getters and setters, which is strange to the eyes, and doesn't have the IDE support like regular POJOs do (stuff like "generate getters and setters based on fields").

Another nuisance happens when you try to make the Template implement the model. By Java limitations, you are forced to create the interface in a separate file:

```
//omitted annotations
public class MyComponent extends PolymerTemplate<MyModel> implements MyModel {
    @Override
    public void setProperty(String property) {
        getModel().setProperty(property);
    }
}
//separate file
public interface interface MyModel extends extends TemplateModel {
        void setProperty(String property);
}
```
Another problem comes with reusability of the model. In some cases you may want to use the same POJO to transfer data from/to the UI, to apply validations and to save data on the database. When using an interface as model, you must write converters to transform the data from the model in to a proper POJO and then use it as desired - which leads to lots of boilerplate code.

The proposal here is to drop the model as interface and allow the usage of POJOs as model:

```
@SomeAnnotation // annotation needed to inform the framework that this class contains @Bound fields
public void MyModel implements Serializable {
    @Bound //or any other name
    private String property;
    
    // setters and getters
}
```
And possibly using the same component as the model (useful for small number, UI-only related properties):
```
@SomeAnnotation // annotation needed to inform the framework that this class contains @Bound fields
public class MyComponent extends Component {
    @Bound
    private String property;
}
```

The usage of interfaces as models was used due to the limitation on Java reflection APIs that only allowed dynamic proxies to be created from interfaces. But that limitation can be avoided when using the [Byte Buddy](http://bytebuddy.net) library.

Byte Buddy is used in well known projects like Hibernate and Mockito to provide their proxy implementations. They provide commercial support (if needed) and currently is becoming the _de facto_ standard for dynamic proxy in Java (replacing cglib and Javaassist).

In summary: use Byte Buddy to provide proxy implementations on top of concrete classes, so the developer doesn't need to create interfaces for model objects.
